### ISSUE TITLE
Is anonymous function ok here? 

### ISSUE THREAD

**Issue by denisergashbaev** [denisergashbaev/learning_plan#2]
https://github.com/denisergashbaev/learning_plan/blob/2f69afb1cf86c56d2ba67b160735f8a66f71995f/ml_engineering/go/exercises/exercise-equivalent-binary-trees.go#L13

I am not quite sure whether it is the right way to do that. I used the inner function in order to be able to close the channel only at the topmost level

**Comment by dexianta** [comment]
yes, not an issue, you can directly do this as well judging from the code	
```
func(t *tree.Tree) {
		if t == nil {
			return
		}
		w(t.Left)
		ch <- t.Value
		w(t.Right)
	}(t)
```

**Comment by denisergashbaev** [comment]
well, I need the declaration of the `var w func(...)` first in order to be able to call `w()` from inside of itself. If I only use the anonymous function I cannot call it recursively
